src/config-validator.ts

```diff
@@ -56,7 +65,16 @@ export const getArray = (name: string): ZodString['_output'][] => {
 	} catch (e) {
 		throw new Error(`Environment variable ${name} is not valid. Please provide an array of strings.`)
 	}
+};
 
+export const getEnum = (name: string, enumList: string[]): string => {
+    const zEnum = z.enum(['', ...enumList]);
+    const value = process.env[name] || '';
+    const stringValue = zEnum.safeParse(value);
+    if (!stringValue.success) {
+        throw new Error('property ' + name + ' in config is not a valid. Value should be one of ' + enumList.toString());
+    }
+    return stringValue.data;
 };
```

Comment:
As written this function generates a new zod enum object every time we invoke it. This works just fine for our use case since we only want to use it for each enum once, but often we will already have our enums defined in a common types directory. So its likely we'd want to take the zod enum as a function argument, and then we can also return the value as an instance of that enum:

```suggestion
export const getEnum = <T extends [string, ...string[]]>(name: string, zEnum: ZodEnum<T>): z.infer<ZodEnum<T>> => {
	const value = process.env[name];
	const stringValue = zEnum.safeParse(value);
	if (!stringValue.success) {
		throw new Error('property ' + name + ' in config is not a valid. Value should be one of ' + enumList.toString());
	}
	return stringValue.data;
};
```

Then can be used like:
```ts
export const AuthStrategy = z.enum(['EGO', 'KEYCLOAK', 'NONE']);
export type AuthStrategy = z.infer<typeof AuthStrategy>;
// authStrategy type will be `"EGO" | "KEYCLOAK" | "NONE"`
export const authStrategy = getEnum('AUTH_STRATEGY', AuthStrategy);
```

Reply:
If you need this to also have a default for when the value is empty, consider adding that as an additional parameter:

```ts
export const getEnum = <T extends [string, ...string[]]>(
	name: string,
	zEnum: ZodEnum<T>,
	defaultValue?: T,
): z.infer<ZodEnum<T>> => {
	const value = process.env[name] || defaultValue;
    // ...
}
```